specs/adding_new_alert_notification_medium.adoc

```diff
@@ -0,0 +1,151 @@
+// vim: tw=79
+
+= Adding a new notification medium
+
+This specification briefs about what needs to be done to plug-in a new
+noticication medium to tendrl's alert delivery system.
+
+
+== Problem description
+
+A new alert delivery channel needs to be added to the Tendrl supported list of
+alert delivery channels.
+
+== Use Cases
+
+A new alert delivery channel needs to be added to the Tendrl supported list of
+alert delivery channels
+
+== Proposed change
+
+To add a new alert delivery channel to Tendrl's notifier the following needs to
+be done:
+
+Every notification plugin needs to extend NotificationPlugin and it is
+automatically injected with alert object in its dispatch_notification method
+
+----
+The NotificationPlugin interface is defined at:
+https://github.com/Tendrl/notifier/blob/master/tendrl/alerting/notification/__init__.py#L28
+----
+
+Following are the functions to be implemented in the new subclass:
+
+* set_destinations: This would let the plugin know to all the destinations
+  specific to the plugin
+* get_alert_destinations: Logic to get refined list of destinations based on
+  current alert notification subscriptions.
+* format_message: If any custom formatting custom to plugin needs to be made
+  in addition to the formatted readable message in alert.
+
+  ----
+  Note: The example for custom formatting the message custom to plugin is as in
+  https://github.com/Tendrl/notifier/blob/master/tendrl/alerting/notification/handlers/mail_handler.py#L86
```

Comment:
Yeah it is to the notifier module but currently refactoring PR by @GowthamShanmugam is under review So I thought of changing once that's in